test/integration/SimplePublishConsumeFlow.test.ts

```diff
@@ -10,92 +10,98 @@ import {
   Datatoken,
   getHash,
   Nft,
-  downloadFile
+  downloadFile,
+  ZERO_ADDRESS
 } from '../../src'
 import { ProviderFees, Erc20CreateParams } from '../../src/@types'
-
-const assetUrl = [
-  {
-    type: 'url',
-    url: 'https://raw.githubusercontent.com/oceanprotocol/testdatasets/main/shs_dataset_test.txt',
-    method: 'GET'
-  }
-]
-const ddo = {
-  '@context': ['https://w3id.org/did/v1'],
-  id: 'did:op:efba17455c127a885ec7830d687a8f6e64f5ba559f8506f8723c1f10f05c049c',
-  version: '4.0.0',
-  chainId: 4,
-  nftAddress: '0x0',
-  metadata: {
-    created: '2021-12-20T14:35:20Z',
-    updated: '2021-12-20T14:35:20Z',
-    type: 'dataset',
-    name: 'dfgdfgdg',
-    description: 'd dfgd fgd dfg dfgdfgd dfgdf',
-    tags: [''],
-    author: 'dd',
-    license: 'https://market.oceanprotocol.com/terms',
-    additionalInformation: {
-      termsAndConditions: true
-    }
-  },
-  services: [
-    {
-      id: 'notAnId',
-      type: 'access',
-      files: '',
-      datatokenAddress: '0xa15024b732A8f2146423D14209eFd074e61964F3',
-      serviceEndpoint: 'https://providerv4.rinkeby.oceanprotocol.com',
-      timeout: 0
-    }
-  ]
-}
+import { Addresses, deployContracts } from '../TestContractHandler'
 
 describe('Simple Publish & consume test', async () => {
   let config: Config
-  let addresses: any
+  let contracts: Addresses
   let aquarius: Aquarius
   let providerUrl: any
+  let publisherAccount: string
+  let consumerAccount: string
+
+  const assetUrl = [
+    {
+      type: 'url',
+      url: 'https://raw.githubusercontent.com/oceanprotocol/testdatasets/main/shs_dataset_test.txt',
+      method: 'GET'
+    }
+  ]
+
+  const ddo = {
+    '@context': ['https://w3id.org/did/v1'],
+    id: '',
+    version: '4.0.0',
+    chainId: 4,
+    nftAddress: '0x0',
+    metadata: {
+      created: '2021-12-20T14:35:20Z',
+      updated: '2021-12-20T14:35:20Z',
+      type: 'dataset',
+      name: 'dataset-name',
+      description: 'Ocean protocol test dataset description',
+      author: 'oceanprotocol-team',
+      license: 'MIT'
+    },
+    services: [
+      {
+        id: 'testFakeId',
+        type: 'access',
+        files: '',
+        datatokenAddress: '0x0',
+        serviceEndpoint: 'https://providerv4.rinkeby.oceanprotocol.com',
+        timeout: 0
+      }
+    ]
+  }
 
   before(async () => {
     config = await getTestConfig(web3)
-    addresses = getAddresses()
     aquarius = new Aquarius(config.metadataCacheUri)
     providerUrl = config.providerUri
   })
 
+  it('Initialize accounts', async () => {
+    const accounts = await web3.eth.getAccounts()
+    publisherAccount = accounts[0]
+    consumerAccount = accounts[1]
+  })
+
+  it('Deploy contracts', async () => {
+    contracts = await deployContracts(web3, publisherAccount)
```

Comment:
Never deploy contracts on integration tests.   Because the other components (aqua/provider/subgraph) are not going to be aware of smartcontract addresses and your tests are either going to fail or they are not relevant